(>11.4) Code style: Mark protected member vars as deprecated - part 2

elkuku opened the pull request:
Continuing from #564
This fixes the warnings on public and protected class vars beginning with an underscore.
Remaining warnings are caused by missing scope modifiers.

elkuku committed:
Code style: deprecate protected class members starting with an
underscore - base

elkuku committed:
Code style: deprecate protected class members starting with an
underscore - cache

elkuku committed:
Code style: deprecate protected class members starting with an
underscore - database

elkuku committed:
Code style: deprecate protected class members starting with an
underscore - document

elkuku committed:
Code style: deprecate protected class members starting with an
underscore - environment

elkuku committed:
Code style: deprecate protected class members starting with an
underscore - error

elkuku committed:
Code style: deprecate protected class members starting with an
underscore - event

elkuku committed:
Code style: deprecate protected class members starting with an
underscore - filesystem

elkuku committed:
Code style: deprecate protected class members starting with an
underscore - html

elkuku committed:
Fix Docblocks

elkuku committed:
Code style: deprecate protected class members starting with an
underscore - installer

elkuku committed:
Code style: deprecate protected class members starting with an
underscore - plugin

elkuku committed:
Code style: deprecate protected class members starting with an
underscore - session

elkuku committed:
Code style: deprecate protected class members starting with an
underscore - updater

elkuku committed:
Code style: deprecate protected class members starting with an
underscore - user

elkuku committed:
Code style: deprecate protected class members starting with an
underscore - utilities

elkuku committed:
Revert scope changes on methods

elkuku committed:
Fix clashing names

elkuku committed:
Fix JObjectTest to expect the "new" methods

joomla-jenkins commented:
Unit testing complete.  There were 0 failures and 0 errors from 1895 tests and 11033 assertions.
Checkstyle analysis reported 10 warnings and 0 errors.

realityking commented:
Please don't mess with JParameter. It's legacy code that will probably be dropped rather soon, no need to change anything there.

jonnsl commented on libraries/joomla/database/table.php:
```diff
@@ -28,6 +28,15 @@
 	 * @var    string
 	 * @since  11.1
 	 */
+	protected $tbl = '';
```
This will probably  cause error in the store method because it will try to store this variable into the database https://github.com/joomla/joomla-platform/blob/staging/libraries/joomla/database/database.php#L783

elkuku committed:
Merge remote branch 'origin/master' into deprecated

elkuku committed:
Don't mess with deprecated classes

elkuku committed:
Member vars starting *without* underscore are not recognized as
"internal fields" by JDatabase::insertObject()

elkuku committed:
Add missing scope modifiers

elkuku commented:
@realityking right. I removed the vars again and changed the `@deprecated` message as in JSimpleXMLElement. Do you know of other classes that are deprecated ?
@jonnsl Good catch - that's kind of a show stopper `:(` I believe JDatabase::insertObject() should use another "method" to identify internal fields.. reverted

joomla-jenkins commented:
Build triggered by changes to the head.

The tests completed but there was a problem parsing the report.
Checkstyle analysis reported 0 warnings and 0 errors.

elkuku committed:
Appearently wrong scope modifier

elkuku committed:
Accidently overwritten

joomla-jenkins commented:
Build triggered by changes to the head.

Unit testing complete.  There were 0 failures and 0 errors from 1971 tests and 11142 assertions.
Checkstyle analysis reported 0 warnings and 0 errors.

elkuku commented:
Look ma, lotsa zeros :)

realityking commented:
JElement and it's subclasses (part of the JParameter system) are also deprecated.

LouisLandry commented:
Let's pull this right after 11.4 has been tagged... what do you think?

elkuku commented:
The CMS might not be "ready" for this change yet.. so - Perfect ;)

elkuku committed:
Merge remote branch 'joomla/master' into deprecated

Conflicts:
	libraries/joomla/document/document.php
	libraries/joomla/updater/updateadapter.php

joomla-jenkins commented:
Build triggered by changes to the head.

Unit testing complete.  There were 0 failures and 0 errors from 1971 tests and 11146 assertions.
Checkstyle analysis reported 0 warnings and 0 errors.

elkuku committed:
Merge remote branch 'joomla/master' into deprecated

Conflicts:
	libraries/joomla/html/parameter.php
	libraries/joomla/html/parameter/element.php
	libraries/joomla/html/parameter/element/category.php
	libraries/joomla/html/parameter/element/componentlayouts.php
	libraries/joomla/html/parameter/element/contentlanguages.php
	libraries/joomla/html/parameter/element/editors.php
	libraries/joomla/html/parameter/element/filelist.php
	libraries/joomla/html/parameter/element/folderlist.php
	libraries/joomla/html/parameter/element/helpsites.php
	libraries/joomla/html/parameter/element/hidden.php
	libraries/joomla/html/parameter/element/imagelist.php
	libraries/joomla/html/parameter/element/languages.php
	libraries/joomla/html/parameter/element/list.php
	libraries/joomla/html/parameter/element/menu.php
	libraries/joomla/html/parameter/element/menuitem.php
	libraries/joomla/html/parameter/element/modulelayouts.php
	libraries/joomla/html/parameter/element/password.php
	libraries/joomla/html/parameter/element/radio.php
	libraries/joomla/html/parameter/element/spacer.php
	libraries/joomla/html/parameter/element/sql.php
	libraries/joomla/html/parameter/element/templatestyle.php
	libraries/joomla/html/parameter/element/text.php
	libraries/joomla/html/parameter/element/textarea.php
	libraries/joomla/html/parameter/element/timezones.php
	libraries/joomla/html/parameter/element/usergroup.php
	libraries/joomla/utilities/simplexml.php

joomla-jenkins commented:
Build triggered by changes to the head.

Unit testing complete.  There were 0 failures and 0 errors from 1991 tests and 11151 assertions.
Checkstyle analysis reported 0 warnings and 0 errors.

chdemko commented:
I think we can find a global solution for these $_name protected variables:

in JObject

```
/**
 * magic get method
 *
 * @param   $propertyName  Property name
 *
 * @return  mixed  the property value
 *
 * @since   12.1
 * @deprecated in 12.3
 */
public function __get($propertyName)
{
    if ($propertyName[0] == '_' && property_exists($this, $newPropertyName = substr($propertyName, 1)))
    {
        JLog::add(get_called_class() . '::$' . $propertyName . ' will be deprecated in Joomla! 12.3. Use ' . get_called_class() . '::$'. $newPropertyName . ' instead.', JLog::WARNING, 'deprecated');
        return $this->$newPropertyName;
    }
    else
    {
        // Trigger an error
        return $this->$propertyName;
    }
}

/**
 * magic set method
 *
 * @param   $propertyName  Property name
 * @param   $value         Property name
 *
 * @return  void
 *
 * @since   12.1
 * @deprecated in 12.3
 */
public function __set($propertyName, $value)
{
    if ($propertyName[0] == '_' && property_exists($this, $newPropertyName = substr($propertyName, 1)))
    {
        JLog::add(get_called_class() . '::$' . $propertyName . ' will be deprecated in Joomla! 12.3. Use ' . get_called_class() . '::$'. $newPropertyName . ' instead.', JLog::WARNING, 'deprecated');
        $this->$newPropertyName = $value;
    }
    else
    {
        $this->$propertyName = $value;
    }
}
```

An in all descendant classes of JObject, we can rename $_name protected variable to $name (this will ensure backward compatibility)

The same can be applied to other "base" class as JCacheController and so on

chdemko commented:
@elkuku Any news on this?

elkuku commented:
No, not really.. If there is a chance to get it merge it might need an update. If not - feel free to close it :(

elkuku commented:
Closing this as it has been split up into several smaller pull requests that simply rename the protected members.

See: Rename protected members in JXXX..
